src/lib/settings.js

```diff
@@ -1,5 +1,33 @@
 'use strict';
 
+const dns = require('dns');
+const logger = require('./logger')('BOOTSTRAP');
+
+// reshape ipv6 addresses to ipv4
+// TODO: Rewrite this async fully so we don't ever have an undefined value
+const raw = process.env.HOSTNAME || process.env.HOST || process.env.IP || 'localhost';
+let host;
+try {
+  logger.warn(`Not using an ip address? YOLO: ${raw}. Attempting to resolve...`);
+  dns.resolve4(raw, undefined, (err, addresses) => {
+    if (err) {
+      logger.error(err);
+      process.exit(1);
+    }
+    const first = addresses.find((record, index) => index === 0);
+    if (first) {
+      logger.info(`Great Scott! ${first} should be an ipv4 address!`);
+      host = first;
+    } else {
+      logger.error(`Still unable to resolve. You're on your own, Ghostrider.`);
```

Comment:
Just curious how the code would be able to reach this line? If you dont have a loop ethernet adapter?

Reply:
Yeah, if DNS can't resolve the ip